caddytest/integration/caddyfile_adapt/log_filter_with_header.txt

```diff
@@ -0,0 +1,148 @@
+localhost {
+	log {
+		output file ./caddy.access.log
+	}
+	log health_check_log {
+		output file ./caddy.access.health.log
+		no_hostname
+	}
+	log general_log {
+		output file ./caddy.access.general.log
+		no_hostname
+	}
+	@healthCheck `header_regexp('User-Agent', '^some-regexp$') || path('/healthz*')`
+	handle @healthCheck {
+		vars access_logger_names "health_check_log,general_log"
```

Comment:
Oof, yeah this is weird. I don't think using a comma-separated value is ideal. Nowhere else in Caddy do we have this pattern.

@mholt I had an idea that we could do string slices for `vars` if we have the config like this:

```
vars access_logger_names [ health_check_log general_log ]
```

Basically if the first and last token are `[` `]` then we can take the list of args in between as a string slice type value.

I think we should make adjustments to `vars` to make it possible to have slices.

Reply:
Hmm; I see what you mean. But it looks really bash-y. I wonder if we could support `["health_check_log", "general_log"]`...

I actually like how clean and simple the CSV is though. :man_shrugging:

Reply:
Now I'm leaning towards adding a new directive `log_name <names...>` which would handle this case for us. That way users don't need to discover the `access_logger_names` var which is kinda a hack, they could just use `log_name` which would be documented like any other directive. This is similar to how `root` works, it's just a directive shortcut for a `vars` handler that sets a variable. I think it would need to be a real handler in JSON though because the `vars` handler still doesn't support an array of values.

Another long-term thing we could consider (out of scope of this PR but just thinking out loud), we could support a syntax like `vars +var_name value`, where `+` means "assume the var is a slice, and append a value to it". Could also add a `-` prefix to clear variable altogether. I dunno. There's room for improvements with the `vars` handler I think. Let's scrap the `[ ]` idea for now :joy:

Reply:
@francislavoie Sorry for the delay! I didn't have access to a computer in the past week. I took a stab at implementing the new `log_name` directive. It seems to work fine with the vars handler, though. Am I missing something?